components/core/cmake/Modules/FindMariaDBClient.cmake

```diff
@@ -20,6 +20,25 @@ include(cmake/Modules/FindLibraryDependencies.cmake)
 find_package(PkgConfig)
 pkg_check_modules(mariadbclient_PKGCONF QUIET "lib${mariadbclient_LIBNAME}")
 
+if(NOT mariadbclient_PKGCONF_FOUND AND APPLE)
+    execute_process(
+        COMMAND brew --prefix mariadb-connector-c
+        RESULT_VARIABLE brew_result
+        OUTPUT_VARIABLE mariadbclient_MACOS_PREFIX
+        ERROR_QUIET
+    )
+    if(NOT brew_result EQUAL 0)
+        message(FATAL_ERROR "mariadb-connector-c not found in Homebrew")
```

Comment:
If the user installs mariadbclient from source, then this error message may make it seem like we're requiring that it's installed via brew instead. How about:

"pkg-config cannot find ${mariadbclient_LIBNAME} and mariadb-connector-c isn't installed via Homebrew."